utils/write-dependencies.js

```diff
@@ -9,107 +9,147 @@ const { join } = require('path');
 const PackageJson = require('./package-json');
 const { resolveConfig } = require('./sf-config');
 
+/**
+ * These are not part of dev-scripts pjson because they depend on dev-scripts and would create a circular dependency
+ * But, if the target repo has the dep, we want to make sure it meets the minimum version.
+ */
+const nonPjsonDependencyMinimums = new Map([
+  ['@salesforce/sf-plugins-core', '^2.4.2'],
+  ['@salesforce/core', '^3.36.0'],
+  ['@salesforce/kit', '^3.0.0'],
+  ['@salesforce/ts-types', '^2.0.1'],
+  ['@oclif/core', '^2.8.2'],
+  ['@salesforce/cli-plugins-testkit', '^3.3.4'],
+  ['@salesforce/plugin-command-reference', '^2.2.8'],
+  ['@oclif/plugin-command-snapshot', '^3.3.13'],
+  ['eslint-plugin-sf-plugin', '^1.15.1'],
+  ['oclif', '^3.8.2'],
+]);
+
+const getVersionNum = (ver) => (ver.startsWith('^') || ver.startsWith('~') ? ver.slice(1) : ver);
+const meetsMinimumVersion = (pjsonDepVersion, devScriptsDepVersion) => {
+  // First remove any carets and tildas
+  const pVersion = getVersionNum(pjsonDepVersion);
+  const dsVersion = getVersionNum(devScriptsDepVersion) ?? nonPjsonDependencyMinimums.get(pjsonDepVersion);
+  // Compare the version in package.json with the dev scripts version.
+  // result === -1 means the version in package.json < dev scripts version
+  // result === 0 means they match
+  // result === 1 means the version in package.json > dev scripts version
+  return pVersion.localeCompare(dsVersion, 'en-u-kn-true') > -1;
+};
+
 module.exports = (projectPath) => {
   const pjson = new PackageJson(projectPath);
 
   const config = resolveConfig(projectPath);
-  const dependencies = pjson.get('devDependencies');
+  const devDependencies = pjson.get('devDependencies');
 
   const added = [];
   const removed = [];
 
-  const getVersionNum = (ver) => (ver.startsWith('^') || ver.startsWith('~') ? ver.slice(1) : ver);
-  const meetsMinimumVersion = (pjsonDepVersion, devScriptsDepVersion) => {
-    // First remove any carets and tildas
-    const pVersion = getVersionNum(pjsonDepVersion);
-    const dsVersion = getVersionNum(devScriptsDepVersion);
-    // Compare the version in package.json with the dev scripts version.
-    // result === -1 means the version in package.json < dev scripts version
-    // result === 0 means they match
-    // result === 1 means the version in package.json > dev scripts version
-    return pVersion.localeCompare(dsVersion, 'en-u-kn-true') > -1;
-  };
+  const devScriptsVersion = (name) =>
+    devScriptsPjson.dependencies?.[name] ||
```

Comment:
should be using `??` here?

Reply:
agree.  Linter is behind here (one of the things I want to use this new tool for!)